app/models/case_log.rb

```diff
@@ -85,6 +85,31 @@ def weekly_net_income
     end
   end
 
+  def weekly_value(field_value)
+    return unless field_value && period
+
+    case period_label
+    when "Every 2 weeks"
+      field_value / 2
+    when "Every 4 weeks"
+      field_value / 4
+    when "Every calendar month"
+      field_value * 12 / 52
+    when "Weekly for 50 weeks"
+      field_value / 52 * 50
+    when "Weekly for 49 weeks"
+      field_value / 52 * 49
+    when "Weekly for 48 weeks"
+      field_value / 52 * 48
+    when "Weekly for 47 weeks"
+      field_value / 52 * 47
+    when "Weekly for 46 weeks"
+      field_value / 52 * 46
+    when "Weekly for 52 weeks"
+      field_value
+    end
```

Comment:
I'm surprised that Rubocop didn't throw this one at you https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/HashLikeCase unless we've disabled it

Reply:
Interesting. I suspect it's because we're returning an operation rather than a static value so your hash would need to contain lambdas?